main.cpp

```diff
@@ -16,19 +17,64 @@ std::ostream &operator<<(std::ostream &os, std::vector<T> const &a) {
 
 // 请修复这个函数的定义：10 分
 template <class T1, class T2>
-std::vector<T0> operator+(std::vector<T1> const &a, std::vector<T2> const &b) {
+auto operator+(std::vector<T1> const &a, std::vector<T2> const &b) {
     // 请实现列表的逐元素加法！10 分
     // 例如 {1, 2} + {3, 4} = {4, 6}
+    using T0=decltype(T1{}+T2{});
+    auto ret=std::vector<T0>{};
+    auto size=std::min(a.size(),b.size());
+    for(int i=0;i<size;i++){
+        ret.push_back(a[i]+b[i]);
+    }
+    return ret;
 }
 
 template <class T1, class T2>
 std::variant<T1, T2> operator+(std::variant<T1, T2> const &a, std::variant<T1, T2> const &b) {
     // 请实现自动匹配容器中具体类型的加法！10 分
+    // 使用临时变量的返回值，提高效率
+    return std::visit([&](const auto &t1,const auto &t2)->std::variant<T1,T2>{
+        return t1+t2;
+    },a,b);
+}
+
+template <class T1, class T2, class T3>
+std::variant<T1,T2> operator+(std::variant<T1,T2> const &a, T3 const &b){
+    // std::varient 和 元素的加法
+    return std::visit([&](const auto &t)->std::variant<T1,T2>{
+        return t+b;
+    },a);
 }
 
+// 非模板可变参数的重载
 template <class T1, class T2>
 std::ostream &operator<<(std::ostream &os, std::variant<T1, T2> const &a) {
     // 请实现自动匹配容器中具体类型的打印！10 分
+    os<<"{";
+    std::visit([&](auto const &ai){
+        for (size_t i = 0; i < ai.size(); i++) {
+            os << ai[i];
+            if (i != ai.size() - 1)
+                os << ", ";
+        }
```

Comment:
```suggestion
        os << ai;
```
已经实现了 vector<T> 的打印函数，不需要再写一遍。

Reply:
收到，已经修改，谢谢老师